processor/src/fast/mod.rs

```diff
@@ -1054,6 +1050,35 @@ impl FastProcessor {
 
         Ok(())
     }
+
+    // TESTING
+    // ----------------------------------------------------------------------------------------------
+
+    /// Convenience sync wrapper to [Self::execute] for testing purposes.
+    #[cfg(any(test, feature = "testing"))]
+    pub fn execute_sync(
+        self,
+        program: &Program,
+        host: &mut impl Host,
+    ) -> Result<StackOutputs, ExecutionError> {
```

Comment:
Should we expose it not for testing purposes as well? The idea is that we may be able to make it work as a truly sync function in the future but that would not require changing the public API.

Also, maybe we keep the name as `execute()` and the current `execute()` would be `execute_async()`?

Reply:
As discussed offline, I left it as is, since these methods now take an `AsyncHost`, and we really would want them to take a `Host`. In the future if we do want to expose a sync version of the execute methods, we can figure out how to do it properly.